### DISCUSSION TITLE
🆕 Project insights now have filtering options

### DISCUSSION THREAD

**Discussion by mmitrik** [community/community#10915] [Projects and Issues]
The project insights Burn up chart 📈 now provides the option to filter the chart using any project field.  You can use this to filter charts just as you filter project views today.  

Try it out and let us know what you think here in this discussion! :octocat: 

_Note, the project insights charts are still in private preview and have not yet rolled out to all organizations._

**Comment by chrisreddington** [comment]
Hi @MMitrik, is there a process to get an organisation into the private preview for this? Using GitHub Projects (beta) in anger, so I'm keen to try this one out :) Thanks

**Reply by mmitrik** [reply]
We're currently working on features to support self-serve onboarding.  In the meantime, if you're interested, email me at my GitHub handle at github.com 😉 

> Using GitHub Projects (beta) in anger

@chrisreddington I'd love to learn more about the "in anger" part - anything in particular causing you painpoints?

**Reply by chrisreddington** [reply]
Thanks @mmitrik! Perhaps "in anger" is a local colloquialism here, but it means I'm using it extensively (so a positive, not negative 😊)

I'm using it primarily for two ways
- managing the backlog for my content creation (podcasts, blogs, etc)
- managing the backlog to refactor my podcast sites's theme as an OSS Hugo theme. 

Both of those are available in this org if it's of interest - https://github.com/orgs/CloudWithChris/projects?type=beta

**Reply by gaocegege** [reply]
@mmitrik Hi I am also interested in the preview. But the mail to your address is blocked because `Message bounced due to organizational settings.`

**Reply by abeaclark** [reply]
I also got the above message @mmitrik -- would love to enroll https://github.com/fermatcommerce/fermat. Cheers.

**Reply by tiarakelly** [reply]
Are there any updates on adding archived project items in the charts?

**Comment by brainwire** [comment]
After completing the task, we archive the issue on the board. How can I view insights on archived issues?

**Reply by mmitrik** [reply]
Thanks for reaching out, @brainwire!  Currently, we are always filtering out archived items in the Burn up chart.  We don't yet have a good way to allow users to include archived items, but we are working on enabling this type of customization in the future.

**Reply by 700software** [reply]
We had to stop using chart because of this :( Created https://github.com/github/feedback/discussions/17280

**Reply by mr-bcm** [reply]
Is there any update on this? There are many situations where being able to include archived items is necessary.

**Reply by yuanchen-zhu** [reply]
Yeah please allow inclusion of archived issues in the chart (which it did before). Otherwise this is a bit useless.

**Reply by peadarDoyle** [reply]
Still appears to be no way to view archived tickets. Which makes insights useless if you archive as part of your Issue management. Would be very nice to see archived ticket to be considered. Though considering the length of time that has passed it is hard to imagine this will ever be tackled. I guess people just move to more mature and feature rich work management solutions.

**Reply by MarkOSullivan94** [reply]
Would love a way to include archived items

**Comment by jasononeil** [comment]
Excited this feature has landed! I love the flexibility / simplicity here. Same filters as you use elsewhere, but otherwise nothing special to customise to get a decent looking chart.

My main feedback is about the date range picker:

<img width="270" alt="image" src="https://user-images.githubusercontent.com/315158/152531170-f912834a-e331-402d-b393-f26cca5a11de.png">

I feel like 90% of the time I'm going to want the dates of one of my iterations. I'd love to pick the iteration to filter the dates rather than add dates manually.

In the product we build our date-picker has some preset options like this, if it gives you any ideas for how to present the UI:

<img width="773" alt="image" src="https://user-images.githubusercontent.com/315158/152531809-6d0f0c74-53b0-4c0f-bceb-7020ef3b562e.png">

**Comment by Nikschavan** [comment]
Anyway to request access and test this feature out for an organisation?

**Reply by Nikschavan** [reply]
Never-mind, I see this is discussed above already

**Comment by bobinrinder** [comment]
@mmitrik 
Our feedback so far:
- We would love to see a burn down report based on an estimate column with an ideal line to get an idea of whether we are ahead or behind progress wise.
Example:
<img width="1103" alt="Screen Shot 2022-03-02 at 1 13 26 PM" src="https://user-images.githubusercontent.com/4162722/156403538-382cae32-479c-4d09-a252-868ecf2fd472.png">

- Would be great to be able to select the columns used for the graph.

**Reply by mmitrik** [reply]
Thanks for the feedback, @bobinrinder 🙇 

> Would be great to be able to select the columns used for the graph.

Yep.  This is a capability we're working towards.  Curios, what columns would you use for this graph?  I'm assuming it's an effort/points/hours column?

cc @omerbensaadon

**Reply by lars-reimann** [reply]
We would like to use story points. Currently, we have defined this as a single select column with values 1, 2, 3, 5, ... but it would be no problem to use a number field instead.

**Reply by isabia** [reply]
I'd like to use the sum of story points per status to do this chart, and that when the item becomes 'Done' that number decrease.

**Reply by jasononeil** [reply]
We’re using a single select with values “small”, “about right” and “too big” 😆

I could switch to numbers if needed but giving a numerical value to single selects would also be nice

**Comment by jfachal** [comment]
we would like to have the option to measure for every specific issue, the time that it spent in every column.
In our case, we have 7 different states over time for an issue: 

`inception > refinement > backlog > todo > inprogress > review > done`

We need to measure the total lead time, but divided in these states. So, we would have important smells .. e.g. if an issue spends much time in refinement, our PO's are not working as expected.

**Reply by gordsport** [reply]
It was possible to measure this with the classics projects API.

Hey @2percentsilk  It doesnt seem to be added for ProjectV2 yet - do you know if we will we get movedcolumnevent for projectV2 ?

**Comment by thibaudcolas** [comment]
Specifically on the filtering – this is working very well! I’m expecting to be using this to filter by assignee, and by sprint. Here’s my favourite example so far – checking our sprint commitment for the current sprint: 

![current-sprint](https://user-images.githubusercontent.com/877585/159091415-7628558a-4bf3-44ee-9db1-5759309039ce.gif)
 
It took me a bit of time to figure out how to configure a chart to do this – I think the configuration options are as clear as they could be if I’m honest, but it would be nice to have more ready-made charts to refer to. Either available out of the box in GitHub, or find a way for people to share their chart configurations informally.

---

Beyond the filtering – well Project Insights is a very cool feature! Now I’d like to assign every single issue / PR in the repository to a project, just so I can create charts to help visualising the overall project’s health. I expanded a bit upon how I could see this being useful [in a separate feedback discussion](https://github.com/github/feedback/discussions/13037), with an example.

**Comment by isabia** [comment]
After a few weeks using it, I realized that what I miss the most is a forecast on burnup/burndown chart.
[Forescast chart example](https://www.youtube.com/watch?v=N07I0mdR7WQ)
Visualization is not good at the beginning of a new sprint, I can't  understand what I'm visualizing.
![image](https://user-images.githubusercontent.com/38692064/159314682-134519ee-f47d-4fe9-9bc9-73ab9a1618ef.png)

**Comment by glazou** [comment]
Hello @mmitrik 
I tried to email you to get access to Burn up chart but the email is rejected. is there a way to get access for [my organization](https://github.com/orgs/operationcanard/projects/15)? Thanks

**Reply by glazou** [reply]
Hello @2percentsilk, could you give me access to the chart feature? Thanks

**Reply by gordsport** [reply]
@glazou - I think that its now open to everyone? 

As far as I have found in using it, you cant see the insights icon at the top right of the project board if your board is set to public.

**Reply by glazou** [reply]
Hi @gordsport  thanks for your answer. I confirm I still don't have access.
The doc still mention it's not available to everyone https://docs.github.com/en/issues/trying-out-the-new-projects-experience/using-insights-with-projects
My project is private and on an organization
I also tried creating a new project like the post below mention and same results.

**Comment by SjoenH** [comment]
Hi, @mmitrik 
Seems like burn-up charts are only available for newly created project boards...? Is there a way to enable it for our existing one as well? 👉 https://github.com/orgs/equinor/projects/87

**Comment by gordsport** [comment]
Hey @mmitrik,

This is awesome, I have been having a play around with it and its super cool. Is there any chance we can have an option to exclude labels from the chart view?

For example I would like to create a bug report using a stacked column to see how the bugs have been raised/closed and their severity. My issues have the labels `bug`, `high` and `low` but also have other labels i.e. `featureA` or `randomLabel`

It would be cool to have a filter:

`label:bug,high,medium,low` that returns the issues (that may also have many other labels), however, it would be cool to have an exclusion option for the chart display. 

Therefore, the **chart** would show the count of items with my bug severity labels (high,med,low) but NOT show the 'excluded' labels i.e. `featureA` `randomLabel`. 

Also anything on cycletimes / review-to-merge time would be cool too.

**Reply by omerbensaadon** [reply]
CC: @2percentsilk who is the product person for this, @mmitrik has moved on from this role

**Reply by 2percentsilk** [reply]
Thanks for the tag, Omer!

Hearing folks loud and clear on cycletimes! No comment on timeline for that yet but it's high on my list based on customer feedback so far, @gordsport. 

As for the filtering ask to make sure I understand, you want to be able to _either_ have inclusion for a chart or exclusion to build a chart? Or is this an AND scenario where your bugs have multiple labels and you want to exclude some of the items that have other labels? Or is it a third option where the ONLY categories you want shown are the ones you put in the inclusion phrase?

Also happy to chat in a call if you're amenable! :)

**Reply by gordsport** [reply]
Hey @2percentsilk,

Im totally happy to jump on a call to discuss. 

In my scenario I have bug issues that have more than just a `bug` label, it would be great to have a chart option to group by label but a subset of all labeled issues returned in the query.

**Comment by Mehdimome** [comment]
در تاریخ سه‌شنبه ۴ اکتبر ۲۰۲۲،‏ ۲۳:۳۸ Brennan Metzelaar <
***@***.***> نوشت:

**Comment by jd-santos** [comment]
Please look at this issue regarding use of the time field in Insights. Only tracking when metadata changed rather than when a ticket was opened/closed makes it difficult to generate useful time-bound charts. I just lost a couple hours to backfilling data on old issues only to see a chart with a huge spike on today's date. 

https://github.com/community/community/discussions/10915#discussioncomment-4180731

**Comment by martruzg** [comment]
Hi,
I just want to add some notices regarding Insights functionality:

there is no convenient way to make basic Scrum burn-down chart (showing how total estimates change during the iteration);
-- if i choose iterations for x axis and story point for y axis, report just shows all iterations in one place and just change between iterations are calculated, only the final calculation in story points are shown - no info about burn-down of those points.
-- if i move task to other iteration or re-estimate it, no history retains in the previous iteration, all statistics (event if the task was partly done in one iteration and got re-estimated and moved to the next one) moves to the new iteration with the final estimation;
![image](https://github.com/user-attachments/assets/1b2d5789-2384-4ccc-9fc4-bf55b0fc3079)
![image](https://github.com/user-attachments/assets/cff4f0ea-ef70-4328-8306-5b8519fdb3c3)
-- if i choose date for x axis and story points for y axis, report count all tasks that ever been assigned to that iteration with the primary estimated value, no changes or history retains, event if the task was partly done in one iteration and got re-estimated and moved to the next one;
![image](https://github.com/user-attachments/assets/ff2ab4aa-f12a-4cfa-aaf6-281c785fd1e4)

**Reply by yongdol503** [reply]
Hi there! I'm a student studying devops. No matter how hard I looked, I couldn't set time to x-axis in insights. Do I need a paid subscription or beta test to use it?

**Comment by amsmith-pro** [comment]
<img width="615" alt="image" src="https://github.com/user-attachments/assets/b2b028c1-8c01-46c6-9777-7da360945097" />

I created a bunch of issues in February, but don't want the "Open" value to accumulate from their creation date(s) but rather a custom field.

I added a custom "Due Date" date field for my roadmap view. I would like the "Open" line graph on the burn up chart to use the custom "Due Date" field rather than the issue creation date, so I can view how I'm keeping pace with my plan. 

I tend to brainstorm a bunch of tasks I'll need to do and add a bunch of issues, then go back later and add planned start/end dates for far down the line. The default behavior of adding expected value on the creation date is therefore not helpful for me.

Has anyone found a way to do this?

**Reply by nikspyratos** [reply]
+1, the insights when using Time on x axis are too restricted to the Issues flow (open/completed/duplicate) vs custom/projects fields, like Status.

**Comment by slinna** [comment]
@amsmith-pro @nikspyratos if you are open to using a marketplace add-on, I suggest taking a look at the [Analytics & Reports](https://github.com/marketplace/analytics-reports) by Screenful. It allows tracking GitHub project data with [any of your custom fields](https://screenful.com/guide/github/creating-charts-based-on-github-projects-custom-fields), including custom date fields.

Here's a guide for tracking planned vs actual using a custom due date field: 

https://screenful.com/guide/github/how-to-compare-planned-with-actual
